static/js/controllers/inbox.js

```diff
@@ -279,25 +279,11 @@ var feedback = require('../modules/feedback'),
       });
 
       $scope.unreadCount = {};
-      $scope.updateUnreadCount = function() {
-        CountUnreadRecords()
-          .then(function(data) {
-            $scope.unreadCount = data;
-          })
-          .catch(function(err) {
-            $log.error('Error fetching read status', err);
-          });
-      };
-      // wait for db.info to avoid uncaught exceptions: #3754
-      DB().info().then(function() {
-        $scope.updateUnreadCount();
-      });
-      Changes({
-        key: 'inbox-read-status',
-        filter: function(change) {
-          return change.doc.type === 'data_record';
-        },
-        callback: $scope.updateUnreadCount
+      UnreadRecords(function(err, data) {
+        if (err) {
+          return $log.error('Error fetching read status', err);
+        }
+        $scope.unreadCount = data;
```

Comment:
Do you want to set this even if there is an error?

Reply:
What would we set it to? 0? I prefer not updating it because the old value is probably reasonably accurate.

Reply:
Sorry I misread the code, I didn't notice the `return`. No question here.